### ISSUE TITLE
NOT query

### ISSUE THREAD

**Issue by fergiemcdowall** [fergiemcdowall/search-index#104]
Add support for NOTing
```javascript
{
  query:{"*":"apple"},
  not: {"*":"fruit"}
}
```

**Comment by sebnmuller** [comment]
Would that be an "apple AND NOT fruit" "apple OR NOT fruit"? Worth having the option to differentiate?

**Comment by fergiemcdowall** [comment]
I think just "`NOT` fruit" that can then be combined with `AND` or `OR`

**AND NOT**
```Javascript
{
  {
    query:{"*":"apple"},
    not{"*":"fruit"}
  }
}
```

**OR NOT**
```Javascript
{
  {
    query:{"*":"apple"}
  },
  {
    not:{"*":"fruit"}
  }
}
```

**XOR**
```Javascript
{
  {
    query:{"*":"apple"}
  },
  {
    query:{"*":"fruit"}
  }
}
```

Would that make logical sense?

**Comment by fergiemcdowall** [comment]
I just edited the comment above btw

**Comment by sebnmuller** [comment]
Makes sense :) How about the case of a super simple not query along the lines of `{query:{"*":"NOT apple"}}}` ? Perhaps for the sake of a simpler syntax could use something similar to Solr i.e. : `{query:{"*":"-apple"}}}`

**Comment by mewwts** [comment]
My two cents: I don't like having to alter strings to do a query. But making it easy I'm all for!

**Comment by mewwts** [comment]
I also suggest editing the query language all together, and not trying to squeeze it in. It could use some thorough design decisions. I think we discussed this last fall?

**Comment by fergiemcdowall** [comment]
Im all about the thorough design decisions.

I tend to agree that altering strings is a bit of a no-no, but also that its nice to keep the query syntax as small as possible.

hmmm

**Comment by eklem** [comment]
It should be AND NOT at least. Anything else will really confuse the users. Yep, the users are siding with me :-)

**Comment by hems** [comment]
:100: 

i guess that is the main reason why i can't really use the lib at the moment unless there is some nifty workaround for this that would allow me work with 130k entries without many problems

**Comment by fergiemcdowall** [comment]
@hems have you investigated the `filter` functionality?

**Comment by eklem** [comment]
What's the use case @hems ?

**Comment by hems** [comment]
@eklem on my case i have to mix a few queries and not queries together.

for instance,  "find entries that matches "banana" but don't match "apple" and dont match "carrot" and match "grape"

**Comment by fergiemcdowall** [comment]
How about a format like this?:

```javascript
// implicit OR that comprises 3 OR conditions
[
  {  // OR this:
    AND: {'*': ['rolex', 'watch']}, // 'rolex' AND 'watch' in '*' (all) field
    NOT: {'*': ['fake']},           // so rolex AND watch AND NOT fake
    weight: 
  },
  {  // OR this:
    AND: {'*': ['casio', 'watch', 'gold']},
    NOT: {'colour': ['black']}      // NOT 'black' anywhere in the 'colour' field
  },
  {  // OR this:
    NOT: {'*': ['hello', 'kitty']}
  }
]
```

**Comment by eklem** [comment]
:+1:

**Comment by hems** [comment]
🍉

**Comment by fergiemcdowall** [comment]
NOT is now implemented in 0.8.0

Docs should be improved
